File: src/main/java/club/xiaojiawei/hearthstone/strategy/mode/GameplayModeStrategy.java
package club.xiaojiawei.hearthstone.strategy.mode;

import club.xiaojiawei.hearthstone.enums.ModeEnum;
import club.xiaojiawei.hearthstone.listen.PowerFileListen;
import club.xiaojiawei.hearthstone.status.Mode;
import club.xiaojiawei.hearthstone.strategy.ModeStrategy;
import club.xiaojiawei.hearthstone.strategy.PhaseStrategy;
import lombok.extern.slf4j.Slf4j;

/**
 * @author 肖嘉威
 * @date 2022/11/25 12:43
 */
@Slf4j
public class GameplayModeStrategy extends ModeStrategy {



    @Override
    public void intoMode() {

    }

    @Override
    protected void log() {
        Mode.setCurrMode(ModeEnum.GAMEPLAY);
        log.info("切換到" + ModeEnum.GAMEPLAY.getComment());
    }

    @Override
    protected void nextStep() {
        PhaseStrategy.dealing = false;
    }

}

File: Hearthstone-Script/src/main/java/club/xiaojiawei/hsscript/strategy/mode/GameplayModeStrategy.kt
package club.xiaojiawei.hsscript.strategy.mode

import club.xiaojiawei.config.log
import club.xiaojiawei.enums.ModeEnum
import club.xiaojiawei.hsscript.status.Mode.prevMode
import club.xiaojiawei.hsscript.strategy.AbstractModeStrategy
import club.xiaojiawei.hsscript.utils.GameUtil
import club.xiaojiawei.hsscript.utils.SystemUtil.delay

/**
 * 游戏界面
 *
 * @author 肖嘉威
 * @date 2022/11/25 12:43
 */
object GameplayModeStrategy : AbstractModeStrategy<Any?>() {

    override fun wantEnter() {
    }

    override fun afterEnter(t: Any?) {
        if (prevMode == ModeEnum.LOGIN || prevMode == null) {
            log.info { "当前对局不完整，准备投降" }
            GameUtil.surrender()
        }
    }
}
